### ISSUE TITLE
Dashboard profiling timeline is missing many events

### ISSUE THREAD

**Issue by m-harmonic** [ray-project/ray#42496]
### What happened + What you expected to happen

I'm expecting the profiling timeline to always contain the complete record of tasks that executed during the job, but for certain jobs they are missing very large numbers of them. This is causing us a lot of issues as it makes the tool unusable in debugging various issues that the timeline is positioned to address. The only way I can work around this is seemingly to fork Ray, which is a last resort.

I see that there is a hardcoded limit of 10,000 tasks:
https://github.com/ray-project/ray/blob/master/dashboard/state_aggregator.py#L748

Was there a reason aside from performance that this limit exists? Is it possible to remove it, or at least provide a way to get a timeline that is complete, from the API? Thanks

### Versions / Dependencies

2.8.0

### Reproduction script

Repro
1. Generate the timeline for a large job. Examine the contents and verify it does not have all tasks that were executed

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

**Comment by scottsun94** [comment]
@rickyyx Can you advise here?

**Comment by rickyyx** [comment]
Yes, the concern is primarily performance driven (both latency and memory overheads)

You could set the below env vars to store a larger number of tasks before starting ray (do this before ray start):

```
export RAY_task_events_max_num_task_in_gcs=1000000 # 1M of tasks allowed in GCS
export RAY_task_events_max_num_status_events_buffer_on_worker=1000000 # 1M of task events allowed to be buffered on worker
export RAY_MAX_LIMIT_FROM_API_SERVER=100000 # 100K to be sent from dashboard server
export RAY_MAX_LIMIT_FROM_DATA_SOURCE=100000 # 100 K to be sent from GCS
```

If you want the dashboard to load more than 10K, @alanwguo is there a config we could change dynamically here?

**Comment by m-harmonic** [comment]
Thanks for the help on those variables. I also noticed probably having to change `RAY_DASHBOARD_MAX_EVENTS_TO_CACHE` and `RAY_DASHBOARD_MAX_ACTORS_TO_CACHE` as well.

Having a way to change everything dynamically would be great. I got around the issue by changing the source file but that's not a great long term solution for us.
